Wingstop_Recovery_Computer_V1/Wingstop_Recovery_Computer_V1.ino

```diff
@@ -389,137 +460,224 @@ void read_barometer(){//takes barometer reading
+  logFile.print(F("MACH_DELAY:\t"));
+  logFile.print(MACH_DELAY); logFile.println(F(" [millis]\t")); 
+  
+  logFile.print(F("Time:\tAltitude:\tEvents:\t"));
+
+
+
+  
 }
 
 void close_file(){
-  myFile.close(); //closes file
+  logFile.close(); //closes file
 }
 
 void reopen_file(){//closes then reopens the file, saving data up to this point
+  if (logFile){
   close_file();//closes the file
+  }
   open_file();//opens the file
 }
 
 void log_data(){//saves current data to sd card
-  myFile.print("\n");
-  myFile.print(t2); myFile.print(F("\t"));       //logs current time
-  myFile.print(x1); myFile.print(F("\t"));       //logs current position
-  
+  if (logFile) {
+  logFile.print("\n");
+  logFile.print(t_current); logFile.print(F("\t"));       //logs current time
+  logFile.print(x_current); logFile.print(F("\t"));       //logs current position
+  }
 }
 
+int SD_findInt(const __FlashStringHelper * key) {
+  char value_string[30];
+  int value_length = SD_findKey(key, value_string);
+  return HELPER_ascii2Int(value_string, value_length);
+}
 
+float SD_findFloat(const __FlashStringHelper * key) {
+  char value_string[30];
+  int value_length = SD_findKey(key, value_string);
+  return HELPER_ascii2Float(value_string, value_length);
+}
 
+int SD_findKey(const __FlashStringHelper * key, char * value) {
+  File configFile = SD.open("config.txt");
 
-//***********PHYSICS FUNCTIONS***********
+  if (!configFile) {
+    // if the file didn't open, turn on buzzer/LED to indicate file problem
+    turn_on_led(); //Turn on LED
+    turn_on_buzzer(); //Turn on buzzer
+    while(1){ //run while loop forever
+    }
+  }
 
+  char key_string[30];
+  char SD_buffer[30 + 30 + 1]; // 1 is = character
+  int key_length = 0;
+  int value_length = 0;
+
+  // Flash string to string
+  PGM_P keyPoiter;
+  keyPoiter = reinterpret_cast<PGM_P>(key);
+  byte ch;
+  do {
+    ch = pgm_read_byte(keyPoiter++);
+    if (ch != 0)
+      key_string[key_length++] = ch;
+  } while (ch != 0);
+
+  // check line by line
+  while (configFile.available()) {
+    int buffer_length = configFile.readBytesUntil('\n', SD_buffer, 100);
+    if (SD_buffer[buffer_length - 1] == '\r')
+      buffer_length--; // trim the \r
+
+    if (buffer_length > (key_length + 1)) { // 1 is = character
+      if (memcmp(SD_buffer, key_string, key_length) == 0) { // equal
+        if (SD_buffer[key_length] == '=') {
+          value_length = buffer_length - key_length - 1;
+          memcpy(value, SD_buffer + key_length + 1, value_length);
+          break;
+        }
+      }
+    }
+  }
 
+  configFile.close();  // close the file
+  return value_length;
+}
 
-int detect_take_off(){//returns 1 if take off is detected, otherwise 0
-  if(x1 >= TAKEOFF_ALTITUDE){
-    return 1;
-  }
-  else{
-    return 0;
+int HELPER_ascii2Int(char *ascii, int length) {
+  int sign = 1;
+  int number = 0;
+
+  for (int i = 0; i < length; i++) {
+    char c = *(ascii + i);
+    if (i == 0 && c == '-')
```

Comment:
Always, always always put the statement after an `if` in brackets `{ }`!  It is a recipe for bugs not to do so.  It's horrible that the language allows for bracketless ifs! Note: programmers often have strong opinions on style :-)

Reply:
In my defense, I plagiarized this code. lol. I fixed it though.